middlewares/accesslog/logger_test.go

```diff
@@ -165,44 +160,70 @@ func TestLoggerJSON(t *testing.T) {
 	assert.Equal(t, len(jsonData), assertCount, string(logData))
 }
 
-func containsKeys(t *testing.T, expectedKeys []string, data map[string]interface{}) {
-	for key, value := range data {
-		if !contains(expectedKeys, key) {
-			t.Errorf("Unexpected log key: %s [value: %s]", key, value)
-		}
-	}
-	for _, k := range expectedKeys {
-		if _, ok := data[k]; !ok {
-			t.Errorf("the expected key '%s' is not present in the map. %+v", k, data)
-		}
-	}
+func TestNewLogHandlerOutputStdout(t *testing.T) {
+	file, restoreStdout := captureStdout(t)
+	defer restoreStdout()
+
+	config := &types.AccessLog{FilePath: "", Format: CommonFormat}
+	doLogging(t, config)
+
+	written, err := ioutil.ReadFile(file.Name())
+	require.NoError(t, err, "unable to read captured stdout from file")
+	require.NotZero(t, len(written), "expected access log message on stdout")
+	requireValidLogData(t, written)
 }
 
-func contains(values []string, value string) bool {
-	for _, v := range values {
-		if value == v {
-			return true
-		}
-	}
-	return false
+func requireValidLogData(t *testing.T, logData []byte) {
```

Comment:
I understand but:
- it's your main assertion. For me, you don't need to mimic Testify in this case.
- this method contains a mix of require and assert, you cannot guarantee it might abort.